src/Lifecycle/Hook.php

```diff
@@ -75,97 +72,49 @@ public function runsInPhase(Phase $phase): bool
         }
 
         throw new RuntimeException('Hook::validate called on a non-validation hook.');
     }
 
-    public function apply(Container $container, Event $event, State $state): void
+    public function apply(Container $container, Event $event): void
     {
         if ($this->runsInPhase(Phase::Apply)) {
-            app(Wormhole::class)->warp($event, fn () => $container->call($this->callback, $this->guessParameters($event, $state)));
+            app(Wormhole::class)->warp($event, fn () => $this->execute($container, $event));
         }
     }
 
-    public function fired(Container $container, Event $event, StateCollection $states): void
+    public function fired(Container $container, Event $event): void
     {
         if ($this->runsInPhase(Phase::Fired)) {
-            $container->call($this->callback, $this->guessParameters($event, states: $states));
+            $this->execute($container, $event);
         }
     }
 
-    public function handle(Container $container, Event $event, StateCollection $states): mixed
+    public function handle(Container $container, Event $event): mixed
     {
         if ($this->runsInPhase(Phase::Handle)) {
-            return $container->call($this->callback, $this->guessParameters($event, states: $states));
+            return $this->execute($container, $event);
         }
 
         return null;
     }
 
-    public function replay(Container $container, Event $event, StateCollection $states): void
+    public function replay(Container $container, Event $event): void
     {
         if ($this->runsInPhase(Phase::Replay)) {
-            app(Wormhole::class)->warp($event, fn () => $container->call($this->callback, $this->guessParameters($event, states: $states)));
+            app(Wormhole::class)->warp($event, fn () => $this->execute($container, $event));
         }
     }
 
-    protected function guessParameters(Event $event, ?State $state = null, ?StateCollection $states = null): array
+    protected function execute(Container $container, Event $event): mixed
     {
-        $metadata = $event->metadata();
-
-        $parameters = [
-            'e' => $event,
-            'event' => $event,
-            $event::class => $event,
-            (string) Str::of($event::class)->classBasename()->snake() => $event,
-            (string) Str::of($event::class)->classBasename()->studly() => $event,
-            'meta' => $metadata,
-            'metadata' => $metadata,
-            'metaData' => $metadata,
-            'meta_data' => $metadata,
-            Metadata::class => $metadata,
-        ];
-
-        if ($state) {
-            $parameters = [
-                ...$parameters,
-                's' => $state,
-                'state' => $state,
-                $state::class => $state,
-                (string) Str::of($state::class)->classBasename()->snake() => $state,
-                (string) Str::of($state::class)->classBasename()->studly() => $state,
-            ];
-        }
```

Comment:
It's worth noting that Verbs will no longer inject parameters based on name. You need to provide a type hint. We could consider adding that functionality back in, but I think using types is significantly more common and then we don't have to worry about the multitude of naming conventions…